### ISSUE TITLE
Build non-master builds in Travis without using --release

### ISSUE THREAD

**Issue by RobotDisco** [rustodon/rustodon#26]
Ideally, non-master builds would be built in debug mode so we are more empowered to investigate should something break.

Since the CI emits a release candidate when successfully compiling "master", that should be the one place where we produce a release build.

**Comment by RobotDisco** [comment]
It looks like the way to do this involves extracting build steps into a shell script and relying on environment variables.

https://github.com/travis-ci/travis-ci/issues/879

I will leave some time for protests and then try implementing something like this

**Comment by BanjoFox** [comment]
For Aardwolf what we have been doing is adding '#[cfg(debug_assertions)]' to the blocks that we do not want to add in production.  So far it has been working for us :)

Here's a sample route

```
// Route to web root
#[cfg(debug_assertions)]
#[get("/web/<file..>")]
fn webroot(file: PathBuf) -> Result<NamedFile, NotFound<String>> {
    let path = Path::new("web/").join(file);
    NamedFile::open(&path).map_err(|_| NotFound(format!("Bad path: {:?}", path)))
}
```

**Comment by barzamin** [comment]
Closed by #29.
